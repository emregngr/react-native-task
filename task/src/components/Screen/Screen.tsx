import React from "react";
import { View } from "react-native";
import { moderateScale } from "react-native-size-matters";
import styled from "styled-components/native";

import { useThemeContext } from "../../contexts/ThemeProvider/hooks";
import { ScreenProps } from "./screenProps.interface";

const Container = styled(View)<{
  backgroundColor: string;
  padding: number;
}>`
  flex: 1;
  align-items: center;
  justify-content: center;
  background-color: ${(props) => props.backgroundColor};
  padding: ${(props) => moderateScale(props.padding)}px;
`;

export default function Screen({ children }: ScreenProps) {
  const theme = useThemeContext();

  return (
    <Container padding={theme.padding.m} backgroundColor={theme.colors.primary}>
      {children}
    </Container>
  );
}
