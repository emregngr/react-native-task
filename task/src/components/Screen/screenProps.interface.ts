export interface ScreenProps {
  children: React.ReactElement;
}
