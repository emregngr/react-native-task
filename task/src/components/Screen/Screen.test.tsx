import React from "react";
import { Text } from "react-native";

import renderWithProviders from "../../utils/test/renderWithProviders";
import Screen from "./Screen";

describe("<Screen />", () => {
  it("renders correctly", () => {
    const ScreenWithText = () => (
      <Screen>
        <Text>test</Text>
      </Screen>
    );

    const toJson: any = renderWithProviders(<ScreenWithText />).toJSON();

    expect(toJson.props.backgroundColor).toBe("#333B65");

    expect(toJson).toMatchSnapshot();
  });
});
