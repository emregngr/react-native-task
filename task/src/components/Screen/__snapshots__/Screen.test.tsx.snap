// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Screen /> renders correctly 1`] = `
<View
  backgroundColor="#333B65"
  padding={16}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#333B65",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "paddingBottom": 25.142857142857142,
        "paddingLeft": 25.142857142857142,
        "paddingRight": 25.142857142857142,
        "paddingTop": 25.142857142857142,
      },
    ]
  }
>
  <Text>
    test
  </Text>
</View>
`;
